joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -993,7 +1028,7 @@ controller_interface::CallbackReturn JointTrajectoryController::on_activate(
   // those are not nan
   trajectory_msgs::msg::JointTrajectoryPoint state;
   resize_joint_trajectory_point(state, dof_);
-  if (read_state_from_command_interfaces(state))
+  if (dof_ == num_cmd_joints_ && read_state_from_command_interfaces(state))
```

Comment:
Why do we exactly need this condition of `dof_ == num_cmd_joints_`?. I couldn't comprehend it. Sorry

Reply:
`assign_point_from_interface` iterates only over the command interfaces, leaving the entries of joints without command interface NaN. This value is then used for `set_point_before_trajectory_msg` and the future sampling will always give NaN for the joint.

Reply:
right now, if I understand well, the `state_current_`, `state_desired_`, and `last_commanded_state_` have the state updated from the hardware full DoF(including passive joints) and then you check at this condition and update the state object and later override the state current and state desired right?.

My point is, why not override only the part of the commanded joints instead of everything?.

The only point I'm not sure is, if we continue with your current implementation, the future sampling it gives is purely linear or with the specific profile right?. I'm unsure it this fits for all the cases, as there might be a case of nonlinear transmission with passive joints.

Reply:
I changed this a little bit with https://github.com/ros-controls/ros2_controllers/pull/797
I'm not sure what you mean with linear/nonlinear interpolation: the trajectory and sampling of it is identical for passive and active joints. of course, it could be that they have a kinematic relation, but the trajectory class doesn't know it..